impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -327,50 +312,102 @@ fml::StatusOr<RenderTarget> MakeDownsampleSubpass(
     const SamplerDescriptor& sampler_descriptor,
     const DownsamplePassArgs& pass_args,
     Entity::TileMode tile_mode) {
-  ContentContext::SubpassCallback subpass_callback =
-      [&](const ContentContext& renderer, RenderPass& pass) {
-        HostBuffer& host_buffer = renderer.GetTransientsBuffer();
-
-        pass.SetCommandLabel("Gaussian blur downsample");
-        auto pipeline_options = OptionsFromPass(pass);
-        pipeline_options.primitive_type = PrimitiveType::kTriangleStrip;
-        pass.SetPipeline(renderer.GetTexturePipeline(pipeline_options));
-
-        TextureFillVertexShader::FrameInfo frame_info;
-        frame_info.mvp = Matrix::MakeOrthographic(ISize(1, 1));
-        frame_info.texture_sampler_y_coord_scale = 1.0;
-
-        TextureFillFragmentShader::FragInfo frag_info;
-        frag_info.alpha = 1.0;
-
-        const Quad& uvs = pass_args.uvs;
-        BindVertices<TextureFillVertexShader>(pass, host_buffer,
-                                              {
-                                                  {Point(0, 0), uvs[0]},
-                                                  {Point(1, 0), uvs[1]},
-                                                  {Point(0, 1), uvs[2]},
-                                                  {Point(1, 1), uvs[3]},
-                                              });
-
-        SamplerDescriptor linear_sampler_descriptor = sampler_descriptor;
-        SetTileMode(&linear_sampler_descriptor, renderer, tile_mode);
-        linear_sampler_descriptor.mag_filter = MinMagFilter::kLinear;
-        linear_sampler_descriptor.min_filter = MinMagFilter::kLinear;
-        TextureFillVertexShader::BindFrameInfo(
-            pass, host_buffer.EmplaceUniform(frame_info));
-        TextureFillFragmentShader::BindFragInfo(
-            pass, host_buffer.EmplaceUniform(frag_info));
-        TextureFillFragmentShader::BindTextureSampler(
-            pass, input_texture,
-            renderer.GetContext()->GetSamplerLibrary()->GetSampler(
-                linear_sampler_descriptor));
-
-        return pass.Draw().ok();
-      };
-  fml::StatusOr<RenderTarget> render_target =
-      renderer.MakeSubpass("Gaussian Blur Filter", pass_args.subpass_size,
-                           command_buffer, subpass_callback);
-  return render_target;
+  if (pass_args.effective_scalar.x >= 0.5f) {
```

Comment:
Since ui.Images will already have full mip chains generated and `TiledTextureContents::RenderToSnapshot` will passthrough in some cases, I think we can still go down this path to let mip filtering take care of the downsample when the input texture has a mip chain? Otherwise I think that case will regress.

Reply:
Hmm good point! Ill make it conditional based on mip count

Reply:
Done